src/runtime_src/driver/xclng/drm/xocl/userpf/xocl_ioctl.c

```diff
@@ -188,7 +188,7 @@ int xocl_ctx_ioctl(struct drm_device *dev, void *data,
 		goto out;
 	}
 
-	if (bitmap_empty(client->cu_bitmap, MAX_CUS))
+	if (bitmap_empty(client->cu_bitmap, MAX_CUS) && uuid_is_null(&client->xclbin_id))
```

Comment:
i think it's better to add a dedicated boolean to keep track of contexts. it's quite complicated to infer...

Reply:
Yup. Thanks